src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -174,14 +188,15 @@ void SchedulingAlgorithmState_free(SchedulingAlgorithmState *algorithm_state) {
   object_entry *obj_entry, *tmp_obj_entry;
   HASH_ITER(hh, algorithm_state->local_objects, obj_entry, tmp_obj_entry) {
     HASH_DELETE(hh, algorithm_state->local_objects, obj_entry);
-    CHECK(obj_entry->dependent_tasks == NULL);
+    CHECK(obj_entry->dependent_tasks->empty());
+    delete obj_entry->dependent_tasks;
     free(obj_entry);
   }
   /* Free the cached information about which objects are currently being
    * fetched. */
   HASH_ITER(hh, algorithm_state->remote_objects, obj_entry, tmp_obj_entry) {
     HASH_DELETE(hh, algorithm_state->remote_objects, obj_entry);
-    utarray_free(obj_entry->dependent_tasks);
+    delete obj_entry->dependent_tasks;
```

Comment:
In modern C++ code, it is good idea to try to never use `new` or `delete`. It is better to use `std::unique_ptr` or some other smart pointer to handle deallocation in an error-safe way (this is usually called RAII)